_posts/2018-02-07-cockpit-161.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Cockpit 161
+author: pitti
+date: 2018-02-07 10:00
+tags: cockpit linux
+slug: cockpit-161
+category: release
+summary: Cockpit with Virtual Machines, LVM, and Hardware Information improvements
+comments: 'true'
+---
+
+Cockpit is the [modern Linux admin interface](http://cockpit-project.org/). We release regularly.
+Here are the release notes from version 161.
+
+### New VMs can be created on Machines page
+
+The Machines page now has a "Create New VM" button for installing a new libvirt
+virtual machine from a local or remote ISO image. It allows the user to set
+some basic properties such as memory and storage sizes.
+
+![Machines Create new VM](/images/machines-create-iso.png)
+
+Thanks to Dominik Perpeet and suomiy for this feature!
+
+### VMs running in Kubernetes can now be deleted
+
+[Cockpit 160](http://cockpit-project.org/blog/cockpit-160.html) added an
+initial overview of kubevirt Virtual Machines. This page now got a first action
+to delete a running VM.
+
+![kubevirt Delete](/images/kubernetes-kubevirt-delete.png)
+
+Thanks to Marek Libra for this feature!
+
+### Improve LVM volume resizing
+
+Cockpit now knows upfront whether shrinking or growing an LVM logical volume is
+possible, and whether the filesystem needs to be unmounted.  Since shrinking
+and growing are quite different in practice, the Storage page now shows
+separate "Shrink" and "Grow" buttons. See it in action:
+
+<iframe width="960" height="720" src="https://www.youtube.com/embed/M43yTrJ0jzc?rel=0" frameborder="0" allowfullscreen></iframe>
+
+### Add new Hardware Information page
+
+On the System page the "Hardware" name is now a link to a new "Hardware
+information" page. For now this shows information about the system name, form
+factor, CPU, BIOS, and PCI devices. Other types of hardware will be added in
+future versions.
+
+![System Hardware Info](/images/system-hwinfo.png)
+
+### Load Application metadata (Appstream) packages on demand on Debian/Ubuntu
+
+[Cockpit 160](http://cockpit-project.org/blog/cockpit-160.html) started to
+install the [AppStream metadata](https://www.freedesktop.org/wiki/Distributions/AppStream/)
+on demand on the Apps page on Fedora. Debian, Ubuntu, and derivatives use a
+different mechanism for that, which is now being handled by Cockpit.
+
+### Rename cockpit-ovirt package to cockpit-machines-ovirt
+
+The previous package name collided with a package that is already available in
+Red Hat Virtualization Host. This only affects Fedora, as other operating systems
+do not ship the Cockpit ovirt extension.
```

Comment:
Someone at FOSDEM said that CentOS erroneously ships this ... and makes it a required subpackage. Leading to annoyances of installing OVirt client tooling when installing Cockpit.

Reply:
I see how CentOS 7 could ship the cockpit-ovirt package, but our packaging (tools/cockpit.spec) never made any of cockpit-{machines,ovirt,machines-ovirt} a dependency of the `cockpit` metapackage, and we never actually packaged it in the RHEL downstream dist-gits. Also, -ovirt is fairly new, so it's curious that it even already found its way there. Doesn't CentOS use or at least base on the rhel dist-gits? Maybe the CentOS packagers did do a deviation there.

I'm downloading the centos-7 image to verify.

Reply:
On current CentOS 7, `yum search cockpit` does not include ovirt, and installing `cockpit` only pulls in -bridge, -system, and -ws. So as far as I can see, we should be safe here. Maybe that someone used some third-party repo?

Reply:
I can remember there was issue in installing cockpit-ovirt and one another package by default, but this was fixed round Cockpit 155